File: Cpp/1603_Design_Parking_System.cpp
class ParkingSystem {
public:
    ParkingSystem(int big, int medium, int small) : slots(3, 0) {
        slots[0] = big;
        slots[1] = medium;
        slots[2] = small;
    }
    
    bool addCar(int carType) {
        if(slots[carType-1] == 0) return false;
        --slots[carType-1];
        return true;
    }
    
    vector<int> slots;
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */

File: Kotlin/1603_Design_Parking_System.kt
class ParkingSystem(big: Int, medium: Int, small: Int) {
    var spaces = IntArray(3)
    
    init {
        spaces[0] = big
        spaces[1] = medium
        spaces[2] = small
    }

    fun addCar(carType: Int): Boolean {
        if(spaces[carType - 1] == 0) {
            return false
        } else {
            --spaces[carType - 1]
        }
        
        return true
    }

}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */